lib/components/user/monitored-trip/trip-status.js

```diff
@@ -0,0 +1,391 @@
+        showTogglePauseTripButton = true
+        break
+      case 'TRIP_UPCOMING':
+        showTogglePauseTripButton = true
+        showToggleSnoozeTripButton = true
+        break
+      case 'TRIP_ACTIVE':
+        showToggleSnoozeTripButton = true
+        showTogglePauseTripButton = true
+        break
+      default:
+        // trip status not yet calculated, return null
+        return null
+    }
+
+    return (
+      <Panel.Body>
+        {showToggleSnoozeTripButton && (
+          <FooterButton onClick={this.onClickToggleSnoozeTodaysTrip}>
+            {monitoredTrip.snoozed
+              ? <><Glyphicon glyph='play' /> Unsnooze trip analysis</>
+              : <><Glyphicon glyph='pause' /> Snooze for rest of today</>
+            }
+          </FooterButton>
+        )}
+        {showTogglePauseTripButton && (
+          <FooterButton onClick={this.onClickTogglePauseTrip}>
+            {monitoredTrip.isActive
+              ? <><Glyphicon glyph='pause' /> Pause</>
+              : <><Glyphicon glyph='play' /> Resume</>
+            }
+          </FooterButton>
+        )}
+        {showDeleteTripButton && (
+          <FooterButton
+            bsStyle='error'
+            onClick={this.onClickDeleteTrip}
+          >
+            <Glyphicon glyph='trash' /> Delete Trip
+          </FooterButton>
+        )}
+        {showPlanNewTripButton && (
+          <FooterButton onClick={this.onClickPlanNewTrip}>
+            <Glyphicon glyph='search' /> Plan New Trip
+          </FooterButton>
+        )}
+      </Panel.Body>
+    )
+  }
+
+  render () {
+    const { monitoredTrip } = this.props
+    const journeyState = this.getJourneyState()
+    if (!journeyState || !journeyState.matchingItinerary) return null
+
+    let panelBsStyle
+    if (!monitoredTrip.isActive || monitoredTrip.snoozed) {
+      panelBsStyle = undefined
+    } else if (
+      journeyState.tripStatus === 'NO_LONGER_POSSIBLE' ||
+        journeyState.tripStatus === 'NEXT_TRIP_NOT_POSSIBLE'
+    ) {
+      panelBsStyle = 'error'
+    } else if (
+      journeyState.tripStatus === 'TRIP_ACTIVE' ||
+        journeyState.tripStatus === 'TRIP_UPCOMING'
+    ) {
+      panelBsStyle = this.renderRealtimeInfo().panelBsStyle
+    }
+
+    return (
+      <Panel bsStyle={panelBsStyle}>
+        {this.renderHeader()}
+        {this.renderBody()}
+        {this.renderAlerts()}
+        {this.renderFooterButtons()}
+      </Panel>
+    )
+  }
+}
+
+// connect to the redux store
+
+const mapStateToProps = (state, ownProps) => {
+  const dateFormat = getDateFormat(state.otp.config)
+  const longDateFormat = getLongDateFormat(state.otp.config)
+  const timeFormat = getTimeFormat(state.otp.config)
+  return {
+    dateFormat,
+    longDateFormat,
+    timeFormat
+  }
+}
+
+const mapDispatchToProps = {
+  deleteUserMonitoredTrip,
+  planNewTripFromMonitoredTrip,
+  togglePauseTrip,
+  toggleSnoozeTrip
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(TripSummary)
```

Comment:
This file is pretty mondo. I don't see any obvious ways to split it up outside of splitting up the header, body, alerts, and footer buttons, but I think that might be warranted.

Reply:
I also didn't quite like how spaghetti this file was and spent a while thinking about how to best organize it both before you wrote this comment and now again after. I have gone ahead and significantly refactored this component into a hopefully more comprehensible structure.